### ISSUE TITLE
MD5 String Hex Value 

### ISSUE THREAD

**Issue by shubhi704** [Zunawe/md5-c#1]
Hello sir, 
I'm trying to understand the MD5 algorithm through the readme file uploaded by you. Can you please tell me why are you using the hex value of the string "Hello, World!" as "48 65 6c 6f 2c 20 57 6f 72 6c 64 21". According to me, It should be "48 65 6c 6c 6f 2c 20 57 6f 72 6c 64 21". Please let me know the reason behind it. 

Regards,
Shubhi

**Comment by Zunawe** [comment]
You're right, I was forgetting an `l`. The `w` was also uppercase instead of lowercase. And when I wrote the message length in decimal that was for 12 characters instead of 13, but in hexadecimal it was for 13 characters.

I think that commit should have fixed it all though.

**Comment by shubhi704** [comment]
Yeah, Thank you for your response and explaining this MD5 in a good manner.

Have a productive day ahead!

Regards,
Shubhi

**Comment by shubhi704** [comment]
Hi,
I'm trying to implement MD5 using Verilog by understanding your code. But
on analysing your code I found that you are first converting 32-bit message
data into little indian form and then adding that 32-bit message that is
input[j] in your case.
Am I analysed this correctly? Please correct me if I am wrong.
And explain me why we are converting this into little indian form.

Regards
Shubhi

**Comment by Zunawe** [comment]
It's probably machine-specific. The machine I wrote this code on was little endian.

The buffer will always be "big endian" because it's just a large chunk of data. It's a series of bytes. When taking pieces of the buffer to do operations on it, I'm storing those pieces in an array of `uint32_t` words. The computer I wrote this on is little endian, so the 32-bit words needed to be the correct little endian representation.

If the computer running this code were big endian, the little endian conversion would be unnecessary. A more robust implementation would check the endianness of the machine to determine whether to convert.
